contracts/data_storages/RideHailingDisputesDataStorage.sol

```diff
@@ -47,6 +53,56 @@ contract RideHailingDisputesDataStorage is DataStorageBaseContract {
         return disputeData[disputeId];
     }
 
+    function getRideId(
+        uint256 disputeId
+    ) external view internalContractsOnly returns (uint) {
+        return disputeData[disputeId].rideId;
+    }
+
+    function setDisputeResponse (
+        uint256 disputeId
+    ) external internalContractsOnly {
+        disputeData[disputeId].responded = true;
+    }
+
+    function getNumOfDefedantUnresponded( //check if have respond to disputes
+        address defendant
+    ) external view internalContractsOnly returns (uint256) {
+        uint unrespondCounter = 0;
+        for(uint i = 0; i < disputeData.length; i++) {
+            if(disputeData[i].defendant == defendant && disputeData[i].responded == false)
+            unrespondCounter ++;
+        }
```

Comment:
Can we maintain a mapping of address => array<dispute> or uint numOpenDisputes instead? This O(n) scan through all the disputes is gonna be costly